src/ILVerification/src/TypeVerifier.cs

```diff
@@ -98,30 +100,80 @@ public void VerifyInterfaces()
                         MethodDesc resolvedMethod = type.ResolveInterfaceMethodTarget(method);
                         if (resolvedMethod is null)
                         {
-                            VerificationError(VerifierError.InterfaceMethodNotImplemented, Format(type), Format(implementedInterface.DefType), Format(method));
+                            VerificationError(VerifierError.InterfaceMethodNotImplemented, Format(type), Format(implementedInterface.DefType, implementedInterface.InterfaceImplementationHandle), Format(method));
                         }
                     }
                 }
             }
         }
 
-        // Format helpers for future use
-        // The idea is to improve formatting string with tokens value i.e.: 
-        // [Assembly]Class(0x1234) vs current [Assembly]Class
-        // in case somebody will pass '--tokens' switch to ILVerify
         private string Format(EcmaType type)
         {
-            return type.ToString();
+            if (_verifierOptions.IncludeMetadataTokensInErrorMessages)
+            {
+                return string.Format("{0}([{1}]0x{2:X8})", type, type.Module, _module.MetadataReader.GetToken(type.Handle));
+            }
+            else
+            {
+                return type.ToString();
+            }
+        }
+
+        private string Format(DefType defType, InterfaceImplementationHandle interfaceImplementationHandle)
+        {
+            if (_verifierOptions.IncludeMetadataTokensInErrorMessages)
+            {
+
+                return string.Format("{0}([{1}]0x{2:X8})", defType, GetModule(defType), _module.MetadataReader.GetToken(interfaceImplementationHandle));
+            }
+            else
+            {
+                return defType.ToString();
+            }
         }
 
-        private string Format(DefType interfaceTypeDef)
+        private string GetModule(DefType defType)
         {
-            return interfaceTypeDef.ToString();
+            if (defType is MetadataType metadataType)
+            {
+                return metadataType.Module.ToString();
+            }
+
+            Debug.Fail("ModuleNotFound");
+
+            return "ModuleNotFound";
         }
 
         private string Format(MethodDesc methodDesc)
         {
-            return methodDesc.ToString();
+            if (_verifierOptions.IncludeMetadataTokensInErrorMessages)
+            {
+                return string.Format("{0}([{1}]0x{2:X8})", methodDesc, GetModule(methodDesc), _module.MetadataReader.GetToken(((EcmaMethod)methodDesc.GetTypicalMethodDefinition()).Handle));
+            }
+            else
+            {
+                return methodDesc.ToString();
+            }
+        }
+
+        private string GetModule(MethodDesc methodDesc)
+        {
+            if (methodDesc is MethodForInstantiatedType methodForInstantiatedType)
```

Comment:
Similar to the above - this can be just:
```
{
    if (methodDesc.GetMethodDefinition().OwningType is MetadataType metadataType)
    {
        return metadataType.Module.ToString();
    }

    Debug.Fail("ModuleNotFound");
    return "ModuleNotFound";
}
```